network/transport/v2/senders.go

```diff
@@ -46,3 +47,84 @@ func (p *protocol) sendGossipMsg(id transport.PeerID, refs []hash.SHA256Hash) er
 		},
 	}})
 }
+
+func (p *protocol) sendTransactionListQuery(id transport.PeerID, refs []hash.SHA256Hash) error {
+	conn := p.connectionList.Get(grpc.ByConnected(), grpc.ByPeerID(id))
+	if conn == nil {
+		// shouldn't happen
+		return errors.New("no connection available")
+	}
+
+	// there shouldn't be more than a ~650 in there, this will fit in a single message
+	refsAsBytes := make([][]byte, len(refs))
+	for i, ref := range refs {
+		refsAsBytes[i] = ref.Slice()
+	}
+
+	envelop := &Envelope_TransactionListQuery{
+		TransactionListQuery: &TransactionListQuery{
+			Refs: refsAsBytes,
+		},
+	}
+
+	conversation := p.cMan.conversationFromEnvelope(envelop)
+	// todo convert to trace logging
+	log.Logger().Infof("requesting transactions from peer (peer=%s, conversationID=%s)", id, conversation.conversationID.String())
+
+	return conn.Send(p, &Envelope{Message: envelop})
+}
+
+func (p *protocol) sendTransactionList(id transport.PeerID, conversationID conversationID, transactions []*Transaction) error {
+	conn := p.connectionList.Get(grpc.ByConnected(), grpc.ByPeerID(id))
+	if conn == nil {
+		// shouldn't happen
+		return errors.New("no connection available")
+	}
+
+	for _, chunk := range chunkTransactionList(transactions) {
+		if err := conn.Send(p, &Envelope{Message: &Envelope_TransactionList{
+			TransactionList: &TransactionList{
+				ConversationID: conversationID.slice(),
+				Transactions:   chunk,
+			},
+		}}); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// chunkTransactionList splits a large set of transactions into smaller sets. Each set adheres to the maximum message size.
+func chunkTransactionList(transactions []*Transaction) [][]*Transaction {
+	chunked := make([][]*Transaction, 0)
+
+	currentSize := 0
+	newSize := 0
+	startIndex := 0
+	endIndex := 0
+
+	// TODO to be tested in practise
```

Comment:
Using gRPC/Protobuf reflect you can calculate the would-be message size (maybe for a test with some small/large random TXs)

Reply:
that would require message chunking based on trial and error. (does this fit, no too large, this then?, still too large)